components/apimgt/org.wso2.carbon.apimgt.solace/src/main/java/org/wso2/carbon/apimgt/solace/SolaceAdminApis.java

```diff
@@ -141,6 +141,84 @@ public CloseableHttpResponse registerAPI(String organization, String title, Stri
+        HttpClient httpClient = APIUtil.getHttpClient(serviceEndpointURL.getPort(), serviceEndpointURL.getProtocol());
+        HttpPatch httpPatch = new HttpPatch(baseUrl + "/" + organization + "/apis/" + title);
+        httpPatch.setHeader(HttpHeaders.AUTHORIZATION, "Basic " + getBase64EncodedCredentials());
+        httpPatch.setHeader(HttpHeaders.CONTENT_TYPE, "text/plain");
+        JsonNode jsonNodeTree;
+        String jsonAsYaml = null;
+        // convert json to yaml
+        try {
+            jsonNodeTree = new ObjectMapper().readTree(apiDefinition);
+            jsonAsYaml = new YAMLMapper().writeValueAsString(jsonNodeTree);
+        } catch (JsonProcessingException e) {
+            log.error(e.getMessage());
+        }
+        //add definition to request body
+        if (jsonAsYaml != null) {
+            StringEntity params = null;
+            try {
+                params = new StringEntity(jsonAsYaml);
+            } catch (UnsupportedEncodingException e) {
+                log.error(e.getMessage());
+            }
+            httpPatch.setEntity(params);
+            try {
+                return APIUtil.executeHTTPRequest(httpPatch, httpClient);
+            } catch (IOException | APIManagementException e) {
+                log.error(e.getMessage());
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Update API Product in Solace using AsyncAPI Definition
+     *
+     * @param organization           name of the Organization
+     * @param environment            name of the Environment
+     * @param apiProductName         name of the API product
+     * @param apiNameForRegistration name of the Solace API product
+     * @param aai20Document          Async definition of the Solace API
+     * @return CloseableHttpResponse of the POST call
+     */
+    public CloseableHttpResponse updateAPIProductDefinition(String organization, String environment,
+                                                            Aai20Document aai20Document, String apiProductName,
+                                                            String apiNameForRegistration) {
+        URL serviceEndpointURL = new URL(baseUrl);
+        HttpClient httpClient = APIUtil.getHttpClient(serviceEndpointURL.getPort(), serviceEndpointURL.getProtocol());
+        HttpPatch httpPatch = new HttpPatch(baseUrl + "/" + organization + "/apiProducts/" + apiProductName);
+        httpPatch.setHeader(HttpHeaders.AUTHORIZATION, "Basic " + getBase64EncodedCredentials());
+        httpPatch.setHeader(HttpHeaders.CONTENT_TYPE, "application/json");
+        //Create Request Body for Patch
+        org.json.JSONObject requestBody = buildAPIProductRequestBody(aai20Document, environment, apiProductName,
+                apiNameForRegistration);
+        //PATCH request body should not contain any additional properties.
+        requestBody.remove("name");
+        try {
+            StringEntity params2 = new StringEntity(requestBody.toString());
```

Comment:
Shall we use a meaningful variable name here instead of using params2?

Reply:
Fixed https://github.com/wso2/carbon-apimgt/pull/11001/commits/9c9fe10fbb271f44e424d8a7efc0a76be0fc7b09